src/main.cpp

```diff
@@ -815,7 +822,14 @@ int main(int argc, char**argv)
       http_af = AF_INET6;
     }
 
-    chronos_callback_addr = options.ralf_hostname;
+    if (options.chronos_ralf_callback_uri != "")
+    {
+      chronos_callback_addr = options.chronos_ralf_callback_uri;
```

Comment:
Can you put port_str on the end of this? That way the two callback_uri options can have the same format.